.github/workflows/pythonpackage.yml

```diff
@@ -12,8 +12,10 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
```

Comment:
Do we need the `fetch-depth: 0` ? The default fetch-depth for the prior v2 is 1 and if we can make this work with a fetch-depth of 1 it would be much more performant.

Reply:
In the code changes section here https://github.com/elastic/detection-rules/pull/3727#issue-2327824708 we have mentioned why we need fetch-depth:0 , we already do this for lock versions workflows.

Reply:
Synced in DM, will try this trick